src/utilities/tree_sitter_utilities.rs

```diff
@@ -79,17 +80,23 @@ pub(crate) fn get_all_matches_for_query(
     // If `recursive` it allows matches to the subtree of self (Node)
     // Else it ensure that the query perfectly matches the node (`self`).
     if recursive || range_matches_self {
-      let mut replace_node_range = captured_node_range;
-
+      let mut final_replace_node_range = captured_node_range;
       if let Some(replace_node_name) = &replace_node {
-        if let Some(r) = get_range_for_replace_node(query, &query_matches, replace_node_name) {
-          replace_node_range = r;
+        if let Some(r) =
+          get_range_for_replace_node(query, &query_matches, replace_node_name, replace_node_idx)
+        {
+          final_replace_node_range = r;
```

Comment:
Not sure why this is `final_replace_node_range`, if we are pushing it on every iteration through the loop, rather than just the last one... what am I missing here?

Reply:
aah its just bad naming practice on my end :| it should be just `replace_node_range`. I was doing some changes, renamed the var, forgot to un-rename the changes when i undid those changes:|